src/hooks/use-github-user.ts

```diff
@@ -0,0 +1,22 @@
+import useSWR from 'swr';
+import { User } from '../types/github';
+
+export function useGithubUser(username?: string | null) {
+  const {
+    data: user,
+    isLoading,
+    error,
+  } = useSWR(
```

Comment:
This data won't change often. The default is 2s I think. Could increase the `dedupingInterval`.